tests/test_mailtrap_backend.py

```diff
@@ -0,0 +1,299 @@
+        )
+
+    def test_send_email(self):
+        """Test sending a basic email"""
+        response = self.message.send()
+        self.assertEqual(response, 1)
+        self.assert_esp_called("https://send.api.mailtrap.io/api/send")
+
+    def test_send_with_attachments(self):
+        """Test sending an email with attachments"""
+        self.message.attach("test.txt", "This is a test", "text/plain")
+        response = self.message.send()
+        self.assertEqual(response, 1)
+        self.assert_esp_called("https://send.api.mailtrap.io/api/send")
+
+    def test_send_with_inline_image(self):
+        """Test sending an email with inline images"""
+        image_data = sample_image_content()  # Read from a png file
+
+        cid = attach_inline_image(self.message, image_data)
+        html_content = (
+            '<p>This has an <img src="cid:%s" alt="inline" /> image.</p>' % cid
+        )
+        self.message.attach_alternative(html_content, "text/html")
+
+        response = self.message.send()
+        self.assertEqual(response, 1)
+        self.assert_esp_called("https://send.api.mailtrap.io/api/send")
+
+    def test_send_with_metadata(self):
+        """Test sending an email with metadata"""
+        self.message.metadata = {"user_id": "12345"}
+        response = self.message.send()
+        self.assertEqual(response, 1)
+        self.assert_esp_called("https://send.api.mailtrap.io/api/send")
+
+    def test_send_with_tag(self):
+        """Test sending an email with one tag"""
+        self.message.tags = ["tag1"]
+        response = self.message.send()
+        self.assertEqual(response, 1)
+        self.assert_esp_called("https://send.api.mailtrap.io/api/send")
+
+    def test_send_with_tags(self):
+        """Test sending an email with tags"""
+        self.message.tags = ["tag1", "tag2"]
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+    def test_send_with_template(self):
+        """Test sending an email with a template"""
+        self.message.template_id = "template_id"
+        response = self.message.send()
+        self.assertEqual(response, 1)
+        self.assert_esp_called("https://send.api.mailtrap.io/api/send")
+
+    def test_send_with_merge_data(self):
+        """Test sending an email with merge data"""
+        self.message.merge_data = {"to@example.com": {"name": "Recipient"}}
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+    def test_send_with_invalid_api_token(self):
+        """Test sending an email with an invalid API token"""
+        self.set_mock_response(status_code=401, raw=b'{"error": "Invalid API token"}')
+        with self.assertRaises(AnymailAPIError):
+            self.message.send()
+
+    @unittest.skip("TODO: is this test correct/necessary?")
+    def test_send_with_recipients_refused(self):
+        """Test sending an email with all recipients refused"""
```

Comment:
This is probably unnecessary, since Mailtrap doesn't seem to provide per-recipient send status. 

Mailtrap's error documentation isn't all that descriptive, so we might need to test this. If you try to send a message with two recipients, both on your suppression list (or previous bounces), what happens? (Send API success and later webhook reject? Or send API error?)

Similarly, if you send with one valid recipient and one suppressed recipient, what happens?

(I can do some investigation here if that would be helpful.)